### DISCUSSION TITLE
Use UART from NSPanel

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#18304] [Q&A]
Hello!!!
Is it possible to use esp uart from NSPanel in *.be driver? I want to connect RS-485 transmitter to esp uart (whitch is used to flash nspanel) and use it from my *.be driver to connect it otherside to main controller.

**Comment by gonzzzales** [comment]
In init function i see initialisation for uart, connected to Nextion screen to flash it.

def init()
        log("NXP: Initializing Driver")
        self.ser = serial(17, 16, 115200, serial.SERIAL_8N1)
        self.sendnx('DRAKJHSUYDGBNCJHGJKSHBDN')
        self.sendnx('rest')
        self.flash_mode = 0
        self.screeninit()
        #self.sendnx("page Main")
        #self.sendnx("Splash.info.txt=\"ver."+self.VERSION+"\"")
    end

The ESP8266 has two hardware UARTS (Serial ports): UART0 on pins 1 and 3 (TX0 and RX0 resp.)
Is it possible to write something like this?
self.ser1 = serial(1, 3, 115200, serial.SERIAL_8N1)

**Reply by sfromis** [reply]
Yes. Do not think of UART as specific pins, you can have serial on your pins of choice.
